robot/impl/local_robot_test.go

```diff
@@ -4239,3 +4239,104 @@ func TestMaintenanceConfig(t *testing.T) {
 		test.That(t, sensorUnBlocked, test.ShouldNotBeNil)
 	})
 }
+
+func TestRemovingOfflineRemotes(t *testing.T) {
+	logger := logging.NewTestLogger(t)
+	ctx := context.Background()
+	cfg := config.Config{
+		Remotes: []config.Remote{
+			{
+				Name:     "remoteOffline",
+				Insecure: true,
+				Address:  "123.123.123.123",
+			},
+		},
+	}
+	r := setupLocalRobot(t, ctx, &cfg, logger)
+	time.Sleep(2 * time.Second)
+	localRobot := r.(*localRobot)
+	remotes := localRobot.manager.resources.FindNodesByAPI(client.RemoteAPI)
+	test.That(t, remotes[0].Name, test.ShouldEqual, "remoteOffline")
+	r.Reconfigure(ctx, &config.Config{})
+	time.Sleep((2 * time.Second))
```

Comment:
if a wait is absolutely necessary, would try to use testutils.WaitForAssertion instead, since that'll check repeatedly and exit when condition is true